### PR TITLE
add rate limited url to http requests

### PR DESCRIPTION
**Pull request by jaxxstorm**
# Description

Fixes # (issue)

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

**Comment by jaxxstorm** [comment]
bors try

File: sdk/go/common/workspace/plugins.go [hunk 1]
```diff
@@ -422,7 +422,7 @@ func (source *githubSource) getHTTPResponse(
 		addAuth = " You can set GITHUB_TOKEN to make an authenticated request with a higher rate limit."
 	}
 
-	logging.Errorf("GitHub rate limit exceeded%s%s", tryAgain, addAuth)
+	logging.Errorf("GitHub rate limit exceeded for host%s%s%s", req.URL.Host, tryAgain, addAuth)
```
Comment (Frassle) [inline sdk/go/common/workspace/plugins.go:425]: Why just the host? Be more useful to print the whole URL here wouldn't it?

Comment (jaxxstorm) [reply sdk/go/common/workspace/plugins.go:425]: I think this is the full url, but happy to reassess?

Comment (Frassle) [reply sdk/go/common/workspace/plugins.go:425]: https://pkg.go.dev/net/url#URL
> Host        [string](https://pkg.go.dev/builtin#string)    // host or host:port

I think we just want `req.URL` here to just get the result of `String()` here which will include the path.

**Comment by jaxxstorm** [comment]
bors try

File: sdk/go/common/workspace/plugins.go (hunk 1 above)
Comment (jaxxstorm) [reply sdk/go/common/workspace/plugins.go]: @Frassle fixed!

File: sdk/go/common/workspace/plugins.go [hunk 2]
```diff
@@ -432,7 +432,7 @@ func (source *githubSource) getHTTPResponse(
 		addAuth = " You can set GITHUB_TOKEN to make an authenticated request with a higher rate limit."
 	}
 
-	logging.Errorf("GitHub rate limit exceeded%s%s", tryAgain, addAuth)
+	logging.Errorf("GitHub rate limit exceeded for host%s%s%s", req.URL, tryAgain, addAuth)
```
Comment (Frassle) [inline sdk/go/common/workspace/plugins.go:435]: "for host"? maybe just "for".

"iGtHub rate limit exceeded for host https://api.github.com/repos/pulumi/pulumi-mock-latest/releases/latest" vs
"GitHub rate limit exceeded for https://api.github.com/repos/pulumi/pulumi-mock-latest/releases/latest"

**Pull request merged**